pyproject.toml

```diff
@@ -1,3 +1,41 @@
+[dependency-groups]
+docs = [
+    "sphinx", "sphinx-copybutton", "sphinx-issues", "furo", "sphinx-design",
+    # required for testing modules to load
+    "responses",
+    # required for the PR pre-build step in RTD
+    "scriv",
+]
+test = [
+    "coverage",
+    "pytest", "pytest-xdist", "pytest-randomly", "flaky",
+    "responses",
+]
+test-mindeps = [
+    {include-group = "test"},
+    # this version of requests is newer than the minimum in package data
+    # however, if we pin back to 2.19.1, we break 'responses' behaviors used by the
+    # testsuite
+    "requests==2.22.0",
+    "pyjwt==2.0.0",
+    "cryptography==3.3.1",
+    "typing_extensions==4.0",
+]
+typing = [
+    "mypy",
+    "types-docutils",
+    "types-jwt",
+    "types-requests",
+    "typing-extensions>=4.0",
+    # although 'responses' is provided by the test requirements, it also
+    # must be installed for proper type-checking on globus_sdk._testing
+    "responses",
+]
+typing-mindeps = [
+    {include-group = "typing"},
+    "typing-extensions==4.0",
+]
+
```

Comment:
I just looked at this in the GitHub render. Should this table come after the `[project]` and `[tools.setuptools]` tables? Maybe push to the bottom of the file?

Reply:
I'm inclined to have this come after -- at least -- `[build-system]`. It's not tool-specific, so it seems fine to have it above the tool-specific sections. :+1:

Reply:
I put it after the `[project]` table. I'm still fussing about "is it right?! is it right?!" internally, but IMO we can live with this and move it around later if we dislike it.